ionoscloud/resource_certificate_manager_provider.go

```diff
@@ -0,0 +1,199 @@
+	}
+	if err := cert.SetProviderData(d, response); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func providerRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CertManagerClient
+	providerID := d.Id()
+	location := d.Get("location").(string)
+	provider, apiResponse, err := client.GetProvider(ctx, providerID, location)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error while fetching auto-certificate provider with ID: %v, error: %w", providerID, err))
+	}
+	log.Printf("[INFO] Successfully retrieved auto-certificate provider with ID: %v, provider info: %+v", providerID, provider)
+	if err := cert.SetProviderData(d, provider); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func providerUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CertManagerClient
+	providerID := d.Id()
+	location := d.Get("location").(string)
+
+	providerUpdateData := cert.GetProviderDataUpdate(d)
+	provider, _, err := client.UpdateProvider(ctx, providerID, location, *providerUpdateData)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while updating auto-certificate provider with ID: %v, error: %w", providerID, err))
+	}
+	if err = utils.WaitForResourceToBeReady(ctx, d, client.IsProviderReady); err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while checking the update status for the auto-certificate provider with ID: %v, error: %w", providerID, err))
+	}
+	if err := cert.SetProviderData(d, provider); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func providerDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CertManagerClient
+	providerID := d.Id()
+	location := d.Get("location").(string)
+	apiResponse, err := client.DeleteProvider(ctx, providerID, location)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error while deleting auto-certificate provider with ID: %v, error: %w", providerID, err))
+	}
+	err = utils.WaitForResourceToBeDeleted(ctx, d, client.IsProviderDeleted)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("deletion check failed for auto-certificate provider with ID: %v, error: %w", providerID, err))
+	}
+	return nil
+}
+
+func providerImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(services.SdkBundle).CertManagerClient
+	parts := strings.Split(d.Id(), ":")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid import ID: %v, expected ID in the format: '<location>:<provider_id>'", d.Id())
+	}
+	location := parts[0]
+	// TODO -- Modify the name of this constant
```

Comment:
resolve TODO or remove it if no longer needed

Reply:
@cbeti-ionos done in ec747230649ad96067a92e1d3b11d469d3eda2f1